### PR TITLE
[WIP] Drop node 10 and webpack 4

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
refactor

**Did you add tests for your changes?**
removed tests for v4.
**If relevant, did you update the documentation?**
Not yet
**Summary**
 Drop node 10 and webpack 4

**Does this PR introduce a breaking change?**
YES

**Other information**
No

### CODE REVIEW

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -1886,103 +1781,16 @@ class WebpackCLI {
...
-            } else if (
-                !configOptions.mode &&
-                process.env &&
-                process.env.NODE_ENV &&
-                (process.env.NODE_ENV === "development" ||
-                    process.env.NODE_ENV === "production" ||
-                    process.env.NODE_ENV === "none")
-            ) {
-                configOptions.mode = process.env.NODE_ENV;
-            }
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/webpack-cli.js:1922]: We should keep these logic, for `procee.env.NODE_ENV`

**Review by alexander-akait — COMMENTED** [review]
Also now we need `--entry-reset` before using `--entry`, it is fine and it is breaking change, some tests need to be rewritten

**Comment by snitin315** [comment]
I will update 👍🏻

**Comment by snitin315** [comment]
Let's rebase after #2833

**Comment by alexander-akait** [comment]
Done here https://github.com/webpack/webpack-cli/pull/2879, also we need simplify our logic in many method, I will do it, anyway thanks for PR

**Pull request closed without merging**